Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs

```diff
@@ -947,11 +947,11 @@ void UpdateToolbar()
 			if (bar == null)
 				return;
 
-			bool isNavigated = NavigationPageController.StackDepth > 1;
 			bar.NavigationIcon = null;
 			Page currentPage = Element.CurrentPage;
+			bool isFirstPage = _fragmentStack.IndexOf(GetPageFragment(currentPage)) == 0;
```

Comment:
`GetPageFragment` enumerates `_fragmentStack` then `IndexOf` enumerates `_fragmentStack`
Could you make a separate method for better optimization?
Or
```suggestion
			bool isFirstPage = _fragmentStack.Count > 0 && (_fragmentStack[0] as FragmentContainer)?.Page == currentPage;
```